gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/ReceivedMessage.java

```diff
@@ -0,0 +1,168 @@
+
+    private final String subscription;
+    private final String ackId;
+    private final PubSub pubsub;
+    private final BuilderImpl delegate;
+
+    private Builder(String subscription, String ackId, PubSub pubsub, BuilderImpl delegate) {
+      this.subscription = subscription;
+      this.ackId = ackId;
+      this.pubsub = pubsub;
+      this.delegate = delegate;
+    }
+
+    @Override
+    Builder id(String id) {
+      delegate.id(id);
+      return this;
+    }
+
+    @Override
+    public Builder payload(byte[] payload) {
+      delegate.payload(payload);
+      return this;
+    }
+
+    @Override
+    public Builder attributes(Map<String, String> attributes) {
+      delegate.attributes(attributes);
+      return this;
+    }
+
+    @Override
+    public Builder addAttribute(String name, String value) {
+      delegate.addAttribute(name, value);
+      return this;
+    }
+
+    @Override
+    public Builder removeAttribute(String name) {
+      delegate.removeAttribute(name);
+      return this;
+    }
+
+    @Override
+    public Builder clearAttributes() {
+      delegate.clearAttributes();
+      return this;
+    }
+
+    @Override
+    Builder publishTime(Long publishTime) {
+      delegate.publishTime(publishTime);
+      return this;
+    }
+
+    @Override
+    public ReceivedMessage build() {
+      return new ReceivedMessage(this);
+    }
+  }
+
+  ReceivedMessage(Builder builder) {
+    super(builder.delegate);
+    subscription = checkNotNull(builder.subscription);
+    ackId = checkNotNull(builder.ackId);
+    pubsub = checkNotNull(builder.pubsub);
+    options = pubsub.options();
+  }
+
+  @Override
+  public Builder toBuilder() {
+    return new Builder(subscription, ackId, pubsub, new BuilderImpl(this));
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(options, super.hashCode());
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null || getClass() != obj.getClass()) {
+      return false;
+    }
+    ReceivedMessage other = (ReceivedMessage) obj;
+    return Objects.equals(toPb(), other.toPb()) && Objects.equals(options, other.options);
+  }
+
+  public PubSub pubSub() {
+    return pubsub;
+  }
+
+  public String subscription() {
+    return subscription;
+  }
+
+  public String ackId() {
+    return ackId;
+  }
+
+  public void acknowledge() {
+    pubsub.acknowledge(subscription, ackId);
+  }
+
+  public Future<Void> acknowledgeAsync() {
+    return pubsub.acknowledgeAsync(subscription, ackId);
+  }
+
+  public void modifyAckDeadline(int deadline, TimeUnit unit) {
+    pubsub.modifyAckDeadline(subscription, deadline, unit, ackId);
+  }
+
+  public Future<Void> modifyAckDeadlineAsync(int deadline, TimeUnit unit) {
+    return pubsub.modifyAckDeadlineAsync(subscription, deadline, unit, ackId);
+  }
+
+  private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Avoid prefixing parameters by in, out or inOut. Uses Javadoc to document this behavior.](https://www.codacy.com/app/mziccard/gcloud-java/file/2314341709/issues/source?bid=3123109&fileBranchId=3188953#l157)